### PR TITLE
Let users run `find_bads_muscle` also when no sensor positions are available

### PR DESCRIPTION
**Pull request by sappelhoff**
This is an "issue as PR", that also contains some commits from #12860 that can be ignored.

The main point that I would like to discuss is, that [`find_bads_muscle`](https://mne.tools/stable/generated/mne.preprocessing.ICA.html#mne.preprocessing.ICA.find_bads_muscle) is using three criteria to flag components as bad by muscle activity. However, two of these criteria require sensor positions to work.

I have a dataset without sensor positions available (and no option to add standard positions), and I would still like to run this function, using only the first criterion to flag channels.

In the diff below, I show that this is easily accomplished in principle. In practice I would much rather have **some** guidance in what components **may** be muscle related, than none at all.

What are your thoughts on this?

### CODE REVIEW

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -2001,43 +2007,59 @@ def find_bads_muscle(
...
-        focus_score = expit((focus_dists - 0.65) / 0.1)
-        # smoothnessness is around 150 for muscle and 450 otherwise
-        # so use reversed logistic centered at 300 with 100 slope
-        smoothness_score = 1 - expit((smoothnesses - 300) / 100)
-        # multiply so that all three components must be present
-        scores = slope_score * focus_score * smoothness_score
+
+        # Need sensor positions for the criteria below
+        try:
+            # compute metric #2: distance from the vertex of focus
```
[the remark this answers is not in the document]
Comment (sappelhoff) [reply mne/preprocessing/ica.py:2016]: thanks for the feedback

Yes, that'd be a cleaner solution, and maybe wrap the scoring after the try/except into a small private function.

are you aware of a simple check whether an object has sensor positions?

[the remark this answers is not in the document]
Comment (mscheltienne) [reply mne/preprocessing/ica.py:2016]: `inst.get_montage() is None` or `inst.info["dig"] is None` should do it? I remember other utilities exist, but I can't remember where at the moment..

[the remark this answers is not in the document]
Comment (mscheltienne) [reply mne/preprocessing/ica.py:2016]: Found it: `mne.utils.checck.py: _check_ch_locs(info, picks=None, ch_type=None)

**Comment by sappelhoff** [comment]
@mscheltienne would you mind reviewing this?

**Comment by mscheltienne** [comment]
I'm away until Monday, I can review then if it's still needed.

File: mne/preprocessing/ica.py [hunk 2]
```diff
@@ -2002,11 +2008,32 @@ def find_bads_muscle(
...
+        # Need sensor positions for the criteria below, so return with only one score
+        # if no positions available
+        picks = _picks_to_idx(
+            inst.info, self.ch_names, "all", exclude=(), allow_empty=False
+        )
+        if not _check_ch_locs(inst.info, picks=picks):
+            logger.warning(
+                "No sensor positions found. Scores for bad muscle components are only "
+                "based on the 'slope' criterion."
+            )
```
Comment (larsoner) [inline mne/preprocessing/ica.py:2024]: For this you should use `warn` from `mne.utils`, it does some nice stuff based on the logging level etc.

File: mne/preprocessing/tests/test_ica.py [hunk 3]
```diff
@@ -1520,6 +1520,15 @@ def test_ica_labels():
     ica.find_bads_muscle(raw)
     assert "muscle" in ica.labels_
 
+    # Try without sensor locations
+    raw.set_montage(None)
+    with catch_logging() as log:
+        labels, scores = ica.find_bads_muscle(raw, threshold=0.35)
+    log = log.getvalue()
+    assert "based on the 'slope' criterion" in log
```
Comment (larsoner) [inline mne/preprocessing/tests/test_ica.py:1528]: Then
```suggestion
    with pytest.warns(RuntimeWarning, match="based on the 'slope'"):
        labels, scores = ica.find_bads_muscle(raw, threshold=0.35)
```

**Review by larsoner — COMMENTED** [review]
Otherwise looks good -- FYI I merged in `main` to get the CI feedback but feel free to push over it

**Comment by larsoner** [comment]
Pushed my suggested fix and marked for merge-when-green, thanks in advance @sappelhoff !

**Pull request merged**